site/content/en/reference/fn/run/_index.md

```diff
@@ -319,14 +319,31 @@ kpt fn run DIR/ --fn-path FUNCTIONS_DIR/
 
 Alternatively, scoping can be disabled using `--global-scope` flag.
 
-### Declaring Multiple Functions
+### Multiple Functions Ordering
 
-You may declare multiple functions. If they are specified in the same file
+Functions execution follows a deterministic order and is based on the following set of rules:
+
+* If the functions are specified in the same file
 (multi-object YAML file separated by `---`), they will
-be run sequentially in the order that they are specified.
+be executed sequentially in the order that they are specified.
+* If the functions are specified in multiple files,
+the functions deeper in the file system tree are executed first.
+* If the functions are specified in multiple files but are at same file
+system tree depth, the function in the file
+with shorter lexical file path is executed first.
```

Comment:
This seems like an implementation detail that user's should rely on. In kubectl, naming files 0-XXX and 1-XXX is a huge hack.

Having said that, we'll revisit the entire orchestration design as part of aligning with Kustomize. The only real concern is migration for users depending on this ordering sematnics.